main.nf

```diff
@@ -586,7 +592,7 @@ process markDuplicates {
 
     script:
     """
-    java -Xmx2g -jar \$PICARD_HOME/picard.jar MarkDuplicates \\
+    java -Xmx${params.java_mem} -jar \$PICARD_HOME/picard.jar MarkDuplicates \\
```

Comment:
I think that a command line option is not required here. This argument sets how much memory is assigned to Picard, and we already know how much memory the process has. So better to just use `task.memory`.

Specifically, we want it in GB, so `-Xmx$(task.memory.toGiga()}g`.

We can also go one further and leave a little overhead. Something like this:
```groovy
script:
java_mem = Math.floor( task.memory.toGiga() * 0.9 )
"""
java -Xmx${java_mem}g ....
"""
```